### DISCUSSION TITLE
Automated trading application performance

### DISCUSSION THREAD

**Discussion by brendan1212** [bennycode/coinbase-pro-node#560] [Q&A] [answered]
Hey @bennycode, after thinking over performance for an automated trading app, I'm wondering if you have any advice or insights on how the ticker or level2 websocket streams perform in a cloud-based environment. 

I assume I'll be needing to run a separate/dedicated service for the real-time data collection / candle building module of my application due to the high amount of messages coming in... Have you experienced any issues with too many messages creating a lag in the system where it can't keep up in real-time? Have you found the need to use worker threads or actually spinning up separate services for different tasks? What kind of vCPU/memory might you recommend for this specific module?

Thanks for all your hard work on this project 👍

**Comment by bennycode** [comment, marked as answer]
Hi @pachiraDIG, the requirements on hardware are pretty low. No need to use worker threads. You can easily run a Node.js application that connects with the [WebSocketClient](https://github.com/bennycode/coinbase-pro-node/blob/v3.0.5/src/client/WebSocketClient.ts) and run it on a [Heroku Hobby Dyno](https://www.heroku.com/dynos) for $7/month. I am telling from experience. 😉

Happy to hear that you like this project! 🙂

**Reply by brendan1212** [reply]
This is great news! I very much appreciate the feedback. Looking forward to digging in.

**Reply by bennycode** [reply]
Cool. Wishing you good luck with it. 🍀 Can you accept my answer in this discussion? I would like to see if this works on GitHub. 😊
